File: Level 1/소수 찾기.cpp
/*
1부터 입력받은 숫자 n 사이에 있는 소수의 개수를 반환하는 함수, solution을 만들어 보세요.
*/

#include <string>
#include <vector>
#include <cmath>

using namespace std;

int solution(int n) {
    int answer = 0;
    vector<bool> v(n+1);
    for(int i=0; i<n; i++)
        v.push_back(false);
    v[0] = v[1] = true;
    for(int i=2; i<sqrt(n); i++){
        if(!v[i]){
            for(int j=i+i; j<=n; j+=i)
                v[j] = true;
        }
    }
    for(int i=1; i<=n; i++){
        if(!v[i]) answer++;
    }
    return answer;
}

File: Level 2/소수 찾기.py
'''
한자리 숫자가 적힌 종이 조각이 흩어져있습니다. 흩어진 종이 조각을 붙여 소수를 몇 개 만들 수 있는지 알아내려 합니다.
각 종이 조각에 적힌 숫자가 적힌 문자열 numbers가 주어졌을 때, 
종이 조각으로 만들 수 있는 소수가 몇 개인지 return 하도록 solution 함수를 완성해주세요.
'''

from itertools import permutations
numbers = "17"

def is_prime(x):
    if x == 0 or x == 1:
        return False
    for i in range(2, x):
        if x % i == 0:
            return False
    return True

def solution(numbers):
    answer = 0
    num_list = []
    all_num_list = set()

    for num in numbers:
        num_list.append(num)

    for i in range(1, len(num_list) + 1):
        for set_list in set(permutations(num_list, i)):
            str_num = ''
            for num in set_list:
                str_num += num
            all_num_list.add(int(str_num))
    
    for num in all_num_list:
        if is_prime(num) == True:
            answer += 1

    return answer

print(solution(numbers))